### PR TITLE
beta docs for offer

### PR DESCRIPTION
**Pull request by shalehaha**

### CODE REVIEW

File: beta/experience-platform-extension/offerdecisioning/offer-decisioning-extension.md [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+* IMS organization is enabled for edge decisioning
+* Offers, Activities created
+* Adobe Experience Platform Core and Edge mobile extensions have been implemented in the app
+* Launch UI config is published
+
+## Enable Offer Decisioning in Edge configuration
+
+To enable Offer Decisioning, you need to perform the following steps:
+
+1. In Adobe Experience Platform Launch, navigate to your mobile property and select Edge Configurations from the left panel, then select the configuration needs to be updated.
```
Comment (swarna04) [inline beta/experience-platform-extension/offerdecisioning/offer-decisioning-extension.md:20]: Grammar: ...select the configuration **that** needs to be updated.

Edge Configurations is at the same level as Properties. Why do we need to navigate to the mobile property first? Isn't it the inlet that specifies the source, eg mobile app?

File: beta/experience-platform-extension/offerdecisioning/offer-decisioning-extension.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+
+   Enabled Adobe Experience Platform in your [edge configuration](https://experienceleague.adobe.com/docs/experience-platform/edge/fundamentals/edge-configuration.html?lang=en) and check the `Offer Decisioning` box
```
Comment (swarna04) [inline beta/experience-platform-extension/offerdecisioning/offer-decisioning-extension.md:22]: Fix: Enabled -> Enable

File: beta/experience-platform-extension/offerdecisioning/offer-decisioning-extension.md (hunk 1 above)
Comment (swarna04) [reply beta/experience-platform-extension/offerdecisioning/offer-decisioning-extension.md:20]: Did you mean select the right IMS org first?

File: beta/experience-platform-extension/offerdecisioning/offer-decisioning-extension.md [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+
+![](/Users/jgeng/work/v5/aep-sdks-documentation/.gitbook/assets/offer-decisioning-edge-config.png)
+
+2. Add AEP Offer Decisioniong extension to the project.
```
Comment (swarna04) [inline beta/experience-platform-extension/offerdecisioning/offer-decisioning-extension.md:26]: Typo Decisioniong -> Decisioning

File: beta/experience-platform-extension/offerdecisioning/offer-decisioning-extension.md [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+*Decision scope JSON:*
+
+```json
+{
+  "activityId":"xcore:offer-activity:11cfb1fa93381aca",
+  "placementId":"xcore:offer-placement:1175009612b0100c"
+}
+```
+
+AEP Offer Decisioning extension provides a convinient class for you to construct the Decision Scope.
```
Comment (swarna04) [inline beta/experience-platform-extension/offerdecisioning/offer-decisioning-extension.md:117]: typo `convinient`

File: beta/experience-platform-extension/offerdecisioning/offer-decisioning-extension.md [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+```swift
+let decisionScope = DecisionScope(activityId: "xcore:offer-activity:11cfb1fa93381aca", placementId: "xcore:offer-placement:1175009612b0100c")
+```
+
+{% endtab %}
+{% endtabs %}
+
+### Prefetch offers
+
+AEP Offer Decisioning exension provides seperate APIs to prefetch the offers and retrives the offer contents. This design is aim to help apps to build better user experience for their end uses. We recommand the app to prefetch the offers in advance, so there is no latency when you rerieve the offer contents and reder them.
```
Comment (swarna04) [inline beta/experience-platform-extension/offerdecisioning/offer-decisioning-extension.md:134]: typo `seperate`, `retrives`, `recommand`, `rerieve`, `reder`

File: beta/experience-platform-extension/offerdecisioning/offer-decisioning-extension.md [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+```
+
+{% endtab %}
+{% endtabs %}
+
+
+
+## Listen for offer update
+
+Offer prefetch can not only be triggered the the forementioned `prefetchOffers` API, calling `Edge.sendEvent()` or using Rules are alternative approaches to prefetch offers. AEP Offer Decisioning extension is listening for all the respsonses from Offer Decisioning Services, no matter how the requests are triggered.
```
Comment (swarna04) [inline beta/experience-platform-extension/offerdecisioning/offer-decisioning-extension.md:189]: Fix: ...triggered by the aforementioned...

Comment (swarna04) [reply beta/experience-platform-extension/offerdecisioning/offer-decisioning-extension.md:189]: Typo: `respsonses`

File: beta/experience-platform-extension/offerdecisioning/offer-api.md [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+
+## Public classes
+
+{% tabs %}
+
+
+{% tab title="iOS" %}
+### DecisionScope/**AEPDecisionScope**
+
+This class contains the id of activity and placement, wich is used by the offer decisioning service to propose offers for.
```
Comment (swarna04) [inline beta/experience-platform-extension/offerdecisioning/offer-api.md:132]: typo `wich`

File: beta/experience-platform-extension/offerdecisioning/offer-api.md [hunk 8]
```diff
@@ -0,0 +1,180 @@
...
+    public let characteristics: [String: String]?
+    public let schema: String
+}
+```
+
+
+
+### OfferType/AEPOfferType
+
+An enum indicates the type of an offer, which is a field of the `Offer` class
```
Comment (swarna04) [inline beta/experience-platform-extension/offerdecisioning/offer-api.md:167]: Fix: An enum indicating...

File: beta/experience-platform-extension/offerdecisioning/offer-api.md [hunk 9]
```diff
@@ -0,0 +1,180 @@
...
+
+```swift
+public enum OfferType{
+    case unknown = 0
+    case json = 1
+    case text = 2
+    case html = 3
+    case image = 4
+}
```
Comment (swarna04) [inline beta/experience-platform-extension/offerdecisioning/offer-api.md:176]: Can we not just write 
```
public enum OfferType: Int {
    case unknown
    case json
    case text
    case html
    case image
}
```
Swift will automatically assign 0 to unknown and count upwards from there.

**Review by swarna04 — CHANGES_REQUESTED** [review]

File: beta/experience-platform-extension/offerdecisioning/offer-api.md (hunk 9 above)
Comment (shalehaha) [reply beta/experience-platform-extension/offerdecisioning/offer-api.md:176]: this class needs to be objective-c compatible, i haven't looked much into this but followed the way how we define enums, I guess the explicit int is needed to support obj-c. I didn't include all that tech detail in the docs .

**Review by imironchuk — APPROVED** [review]
Looks good to me

File: beta/experience-platform-extension/offerdecisioning/offer-api.md (hunk 9 above)
Comment (swarna04) [reply beta/experience-platform-extension/offerdecisioning/offer-api.md:176]: Yes that's correct. Enum cases in swift don't have integer values set by default unlike obj-C . So once we define an enum to be of a particular type, say Int we do not need to assign a raw value to each case (Implicit assignment). Also the first case starts with 0 (unless specified) and the rest of the cases count upwards. Anyways, I'll leave it up to you.

**Review by swarna04 — CHANGES_REQUESTED** [review]

**Comment by swarna04** [comment]
Suggested a few more minor edits before we merge this @shalehaha

**Pull request merged**